src/graph_transpiler/webdnn/frontend/keras/converter.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding:utf-8 -*-
+
+"""
+Keras model -> Graph object converters
+Assuming Keras 2.0.4
+
+Currently, the system assumes the model is trained with "data_format" == "channels_last".
+If this is not the case, Flatten layer which follows Convolution have to change the order of variable.
+Convolution implementation is currently assuming variable is NHWC.
+"""
+
+from collections import defaultdict
+from typing import List
+
+import keras
+import keras.backend as K
+import tensorflow as tf
+
+from webdnn.frontend.converter import Converter
+from webdnn.graph.graph import Graph
+from webdnn.graph.order import OrderNC, Order, OrderNHWC
+from webdnn.graph.placeholder import Placeholder
+from webdnn.graph.variable import Variable
+from webdnn.graph.variables.constant_variable import ConstantVariable
+
+
+def _to_list(x):
+    return x if isinstance(x, (list, tuple)) else [x]
+
+
+class KerasConverter(Converter[keras.layers.Layer]):
+    """
+    KerasConverter
+
+    Args:
+        batch_size: input batch size (default=1)
+
+    """
+
+    def __init__(self, batch_size: int = 1):
+        self._input_index_dict = defaultdict(lambda: 0)
+        self._output_index_dict = defaultdict(lambda: 0)
+        self._placeholder_N = Placeholder(label='N', value=batch_size)
+
+    def convert(self, model: keras.models.Model) -> Graph:
+        """
+        Convert kerasmodel into WebDNN IR Graph. Currently, only TensorFlow backend is supported.
+
+        Args:
+            model (`keras.models.Model`): keras model
+
+        Returns:
+            (:class:`~webdnn.graph.graph.Graph`): WebDNN IR Graph
+        """
+        for tf_tensor in model.inputs:
+            # FIXME: this assumption about input tensors' order is too ugly.
+            shape = [self._placeholder_N if d.value is None else d.value for d in tf_tensor.shape]
+            order = OrderNC if len(shape) == 2 else OrderNHWC
```

Comment:
data_format == "channels_first" is not considered.
But in Conv2D converter, it is considered. Seems inconsistent.
There are 3 dimensional input (NTC). See text_generation example.